### PR TITLE
AP_BattMonitor: Fwd battery data to ESC_Telem

### PR DESCRIPTION
**Pull request by magicrub**
Adds ability to forward battery data to ESC Telemetry via new param BATTx_ESC_INDDEX wrapped in AP_BATTERY_ESC_TELEM_OUTBOUND_ENABLED. Repeat of the AP_RPM changes in https://github.com/ArduPilot/ardupilot/pull/24646

I opted to inherit AP_ESC_TELEM and in stead call it directly, like other libraries do for update_rpm(). This required moving update_telem_data() to public which then defeats the purpose of others having to inherit.

### CODE REVIEW

**Comment by magicrub** [comment]
I've tagged DevCallTopic because of AP_BattMonitor_Backend not inheriting AP_ESC_Telem_Backend and opting to move update_telem_data() to public. I'm happy to inherit, it costs 24 bytes of flash so nut much of a difference.. The telem backend just calls the frontend so not really sure what we gain/lose really by inheriting or not in this case (or any case, really).

File: libraries/AP_BattMonitor/AP_BattMonitor_config.h [hunk 1]
```diff
@@ -30,6 +30,10 @@
 #define AP_BATTERY_ESC_ENABLED AP_BATTERY_BACKEND_DEFAULT_ENABLED && HAL_WITH_ESC_TELEM
 #endif
 
+#ifndef AP_BATTERY_ESC_TELEM_OUTBOUND_ENABLED
+#define AP_BATTERY_ESC_TELEM_OUTBOUND_ENABLED AP_BATTERY_BACKEND_DEFAULT_ENABLED && HAL_WITH_ESC_TELEM && (BOARD_FLASH_SIZE > 1024)
```
Comment (peterbarker) [inline libraries/AP_BattMonitor/AP_BattMonitor_config.h:34]:
```suggestion
#define AP_BATTERY_ESC_TELEM_OUTBOUND_ENABLED 0
```

... and put a define in `periph_defaults.h`

Comment (tridge) [inline libraries/AP_BattMonitor/AP_BattMonitor_config.h:34]: make it off by default and add to build_options.py

**Review by tridge — CHANGES_REQUESTED** [review]

File: libraries/AP_BattMonitor/AP_BattMonitor_config.h (hunk 1 above)
Comment (magicrub) [reply libraries/AP_BattMonitor/AP_BattMonitor_config.h]: done, please re-review

Comment (magicrub) [reply libraries/AP_BattMonitor/AP_BattMonitor_config.h]: @peterbarker also added to defaults_periph.h, per your request

**Review by peterbarker — APPROVED** [review]

**Pull request merged**